### PR TITLE
Print summary document

### PR DESCRIPTION
**Pull request by dwhenry**
Requires https://github.com/guidance-guarantee-programme/output-templates/pull/34 to be merged first.

Update Summary Document generator and text elements

Updates:

* output-templates to get address positioning change
* remove first name from greeting
* remove guider organisation from lead paragraph

### CODE REVIEW

File: app/controllers/appointment_summaries_controller.rb [hunk 1]
```diff
@@ -61,4 +69,11 @@ def authenticate_team_leader!
 
     redirect_to :root unless current_user.team_leader?
   end
+
+  def ajax_reponse_paths(appointment_summary)
```
Comment (benlovell) [inline app/controllers/appointment_summaries_controller.rb]: Typo here for 'response'

File: app/views/appointment_summaries/preview.html.erb [hunk 2]
```diff
@@ -11,8 +11,29 @@
...
+               submit_button_text: 'Send', submit_button_classes: submit_button_classes
+      %>
     </div>
   </div>
 </div>
+
+<script>
+  $(".js-save-and-open").click(function(){
+    var w = window.open('<%= generate_appointment_summaries_path %>');
+    $.ajax({
```
Comment (benlovell) [inline app/views/appointment_summaries/preview.html.erb]: Could use the shorthand `$.post` instead and dispense with the explicit `method`.

File: app/views/appointment_summaries/preview.html.erb [hunk 3]
```diff
@@ -11,8 +11,29 @@
...
+      method: 'POST',
+      url: '<%= appointment_summaries_path(format: 'json') %>',
+      data: $(this.form).serialize(),
+      success: function( data )
+      {
+        w.location = data.pdf_path;
+        window.location = data.done_path;
+      }
+    });
+    return false;
```
Comment (benlovell) [inline app/views/appointment_summaries/preview.html.erb]: I'm not JS expert but should we use `preventDefault` here instead?

Comment (benbarnett) [inline app/views/appointment_summaries/preview.html.erb]: Yes much safer – `return false` will stop the entire click event in its tracks, where as `preventDefault` will allow other click listeners to do their thing.

File: app/views/appointment_summaries/preview.html.erb (hunk 2 above)
Comment (benbarnett) [inline app/views/appointment_summaries/preview.html.erb]: 👍

File: app/views/appointment_summaries/preview.html.erb [hunk 4]
```diff
@@ -1,18 +1,42 @@
...
+    var popupWindow = window.open('<%= generate_appointment_summaries_path %>');
+
+    $.post('<%= appointment_summaries_path(format: 'json') %>', $(this.form).serialize())
+      .done(function(data) {
+        popupWindow.location = data.pdf_path;
+        window.location = data.done_path;
+      })
+      .fail(function() {
+        popupWindow.close();
+        alert( "Error saving the summary document. PLease try again." );
```
Comment (benbarnett) [inline app/views/appointment_summaries/preview.html.erb]: Small typo (PL -> Pl)

File: app/views/appointment_summaries/generate.html.erb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<div class="page-header">
+  <h1><%= title 'Generating PDF' %></h1>
```
Comment (benbarnett) [inline app/views/appointment_summaries/generate.html.erb]: Might just be worth adding a quick explainer?

`<p>The summary document will open in a new window.</p>`

**Comment by benbarnett** [comment]
LGTM. We'll need to add the princexml license to heroku too. 👍

File: app/views/appointment_summaries/generate.html.erb (hunk 5 above)
Comment (dwhenry) [inline app/views/appointment_summaries/generate.html.erb]: done. :)

**Comment by dwhenry** [comment]
@benbarnett need to check with you on prince licence requirements for heroku - is it more than just the two environment variables.

**Pull request merged**